store/store.go

```diff
@@ -147,6 +155,24 @@ func (s *store) SaveBlock(blk *block.Block, cert *certificate.BlockCertificate)
 	s.txStore.saveTxs(s.batch, blk.Transactions(), regs)
 	s.txStore.pruneCache(height)
 
+	// Removing old block from prune node store.
+	if s.isPruned {
+		lc := s.lastCertificate()
+		if lc == nil {
+			return
+		}
+
+		pruneH := lc.Height() - s.config.RetentionBlocks()
+		if pruneH <= 0 {
+			return
+		}
+
+		deleted, err := s.pruneBlock(pruneH)
```

Comment:
After calling `pruneBlock` you need the call method `WriteBatch`

Reply:
@Ja7ad  It is called externally.